crates/gitbutler-cli/src/command/vbranch.rs

```diff
@@ -164,16 +194,40 @@ pub fn commit(project: Project, branch_name: String, message: String) -> Result<
     )?)
 }
 
-pub fn branch_by_name(project: &Project, name: &str) -> Result<Stack> {
-    let mut found: Vec<_> = VirtualBranchesHandle::new(project.gb_dir())
-        .list_all_stacks()?
-        .into_iter()
-        .filter(|b| b.name == name)
-        .collect();
+pub fn stack_by_name(project: &Project, name: &str) -> Result<Stack> {
+    let mut found = find_all_stacks_by_name(project, name)?;
     if found.is_empty() {
-        bail!("No virtual branch named '{name}'");
+        bail!("No stack named '{name}'");
     } else if found.len() > 1 {
-        bail!("Found more than one virtual branch named '{name}'");
+        bail!("Found more than one stack named '{name}'");
     }
     Ok(found.pop().expect("present"))
 }
+
+pub fn stack_by_refname(project: &Project, refname: &Refname) -> Result<Option<Stack>> {
+    let mut found = find_all_stacks_by_refname(project, refname)?;
+    if found.is_empty() {
+        return Ok(None);
+    } else if found.len() > 1 {
+        bail!("Found more than one stack with refname '{refname}'");
+    }
+    Ok(Some(found.pop().expect("present")))
+}
+
+fn find_all_stacks_by_name(project: &Project, name: &str) -> Result<Vec<Stack>> {
+    let found: Vec<_> = VirtualBranchesHandle::new(project.gb_dir())
+        .list_all_stacks()?
+        .into_iter()
+        .filter(|b| b.name == name)
+        .collect();
+    Ok(found)
```

Comment:
```suggestion
    VirtualBranchesHandle::new(project.gb_dir())
        .list_all_stacks()?
        .into_iter()
        .filter(|b| b.name == name)
        .collect()
```

Removing `found` makes the `collect()` type perfectly implicit, which overall should be less noisy.